kbchat/kbchat.go

```diff
@@ -580,6 +580,77 @@ func (a *API) GetUsername() string {
 	return a.username
 }
 
+func (a *API) ListChannels(teamName string) ([]string, error) {
+	apiInput := fmt.Sprintf(`{"method": "listconvsonname", "params": {"options": {"topic_type": "CHAT", "members_type": "team", "name": "%s"}}}`, teamName)
+	output, err := a.doFetch(apiInput)
+	if err != nil {
+		return nil, err
+	}
+
+	var channelsList ChannelsList
+	inboxRaw := output.Text()
+	if err := json.Unmarshal([]byte(inboxRaw[:]), &channelsList); err != nil {
+		return nil, err
+	}
+
+	var channels []string
+	for _, conv := range channelsList.Result.Convs {
+		channels = append(channels, conv.Channel.TopicName)
+	}
+	return channels, nil
+}
+
+func (a *API) JoinChannel(teamName string, channelName string) error {
+	apiInput := fmt.Sprintf(`{"method": "join", "params": {"options": {"channel": {"name": "%s", "members_type": "team", "topic_name": "%s"}}}}`, teamName, channelName)
+	_, err := a.doFetch(apiInput)
```

Comment:
oh - humble request to start changing this pattern (at least on the new stuff). there's no reason not to return what keybase gives back from these requests. this bot library never does that, but it should (especially for creating new chat messages, otherwise handling replies to your messages is a pain -- that's an aside). there is potentially useful information in there. i'm not exactly sure what it would be here, but i kinda think we should just ship it back to the user to deal with if they care to.

Reply:
Done. There really isn't much in the response (just ratelimit information) but I added it. And then over time we can gradually change all the functions to include that information.